File: Ontology/ObjectOneOf.cs
using System.Collections.Generic;

namespace Ontology
{
    public class ObjectOneOf: IObjectOneOf
    {
        private readonly IList<object> _individuals;

        public ObjectOneOf(
            params object[] individuals
            )
        {
            _individuals = individuals;
        }

        IList<object> IObjectOneOf.Individuals => _individuals;

        void IClassExpression.Accept(
            IClassExpressionVisitor visitor
            ) => visitor.Visit(this);

        bool IClassExpression.Evaluate(
            IClassMembershipEvaluator evaluator,
            object                    individual
            ) => evaluator.Evaluate(
                this,
                individual);
    }
}

File: Web/ClientApp/src/app/Ontology/ObjectOneOf.ts
import { ClassExpression } from "./ClassExpression";
import { IClassExpressionSelector } from './IClassExpressionSelector';
import { IClassExpressionVisitor } from "./IClassExpressionVisitor";
import { IIndividual } from "./IIndividual";
import { IObjectOneOf } from "./IObjectOneOf";

export class ObjectOneOf
    extends ClassExpression
    implements IObjectOneOf
{
    public constructor(
        public Individuals: IIndividual[]
        )
    {
        super();
    }

    Accept(
        visitor: IClassExpressionVisitor
        )
    {
        visitor.ObjectOneOf(this);
    }

    Select<TResult>(
        selector: IClassExpressionSelector<TResult>
        ): TResult
    {
        return selector.ObjectOneOf(this);
    }
}
